keyboards/cipulot/ec_pro2/ec_switch_matrix.c

```diff
@@ -0,0 +1,182 @@
+/* Copyright 2023 Cipulot
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ec_switch_matrix.h"
+#include "analog.h"
+#include "atomic_util.h"
+#include "print.h"
+
+/* Pin and port array */
+const uint32_t row_pins[]     = MATRIX_ROW_PINS;
+const uint8_t  col_channels[] = MATRIX_COL_CHANNELS;
+const uint32_t mux_sel_pins[] = MUX_SEL_PINS;
+
+static ecsm_config_t config;
+static uint16_t      ecsm_sw_value[MATRIX_ROWS][MATRIX_COLS];
+
+static adc_mux adcMux;
+
+static inline void discharge_capacitor(void) {
+    ATOMIC_BLOCK_FORCEON {
+        writePinLow(DISCHARGE_PIN);
+    }
+}
+static inline void charge_capacitor(uint8_t row) {
+    ATOMIC_BLOCK_FORCEON {
+        writePinHigh(DISCHARGE_PIN);
+        writePinHigh(row_pins[row]);
+    }
+}
```

Comment:
These atomic blocks should probably be removed (the one inside `discharge_capacitor()` does not have anything to protect against, and the one inside `charge_capacitor()` should become a single atomic block that covers both the pin manipulations and the ADC conversion start).

Reply:
Covered in the prev request change.